openssl/src/base64.rs

```diff
@@ -0,0 +1,129 @@
+//! Utilities for base64 coding
+//!
+//! See manual page of [`EVP_EncodeInit`] for more information on the specific base64 variant.
+//!
+//! [`EVP_EncodeInit`]: https://www.openssl.org/docs/man1.1.1/man3/EVP_EncodeInit.html
+use error::ErrorStack;
+use ffi;
+use libc::c_int;
+use cvt_n;
+
+/// Encodes a given block of bytes to base64.
+///
+/// # Panics
+///
+/// Panics if the input length or computed output length
+/// overflow a signed C integer.
+pub fn encode_block(src: &[u8]) -> String {
+    let src_len = usize_to_c_int(src.len()).unwrap();
+    let len = encoded_len(src_len).unwrap();
+    let mut out = Vec::new();
+    out.reserve(len);
+
+    // SAFETY: `encoded_len` ensures space for 4 output characters
+    // for every 3 input bytes including padding and nul terminator.
+    // `EVP_EncodeBlock` will write only single byte ASCII characters.
+    // `EVP_EncodeBlock` will only write to not read from `out`.
+    unsafe {
+        let out_len = ffi::EVP_EncodeBlock(out.as_mut_ptr(), src.as_ptr(), src_len);
+        out.set_len(c_int_to_usize(out_len).unwrap());
```

Comment:
This can just be `out_len as usize`.

Reply:
Indeed. Fixed this and the checked cast in `{de,en}coded_len` that was actually unnecessary as well as we always call this with non-negative values and the arithmetic is checked.